sympy/polys/rings.py

```diff
@@ -3005,3 +3178,295 @@ def sqf_list(f, all=False):
 
     def factor_list(f):
         return f.ring.dmp_factor_list(f)
+
+def cont_prim(p, x):
+    """Returns the content and primitive part of a polynomial with respect to a
+    specified variable."""
```

Comment:
Maybe call this `primitive_wrt`. It should be a `PolyElement` method rather than a function.